From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aswin-mcw <aswin.kalies@multicorewareinc.com>
Date: Thu, 30 Jul 2026 17:09:24 +0530
Subject: [PATCH 096/109] fixup! ci: run Cygwin's testsuite on the MSYS2
 runtime

Start cygserver via the always-present Windows PowerShell 5.1 instead of
pwsh, which is not on the restricted PATH used by this step. Without it
cygserver never started and the SysV IPC tests (semtest, shmtest,
msgtest) failed spuriously.

(cherry picked from commit 9dd0b09e6772f0e5b7d45bb756dd26634eab17c2)
(cherry picked from commit 423369254a2b9a8d82d7017500711ead0438208a)
---
 .github/workflows/build.yaml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.github/workflows/build.yaml b/.github/workflows/build.yaml
index 6d0474475..43b006487 100644
--- a/.github/workflows/build.yaml
+++ b/.github/workflows/build.yaml
@@ -150,7 +150,7 @@ jobs:
           cp $WINSUP/cygserver/cygserver.exe $WINSUP/testsuite/testinst/usr/bin/ &&
           (cd $WINSUP/testsuite/testinst/usr/bin &&
            PATH=$PWD:$WINDIR/system32
-           pwsh -command '& .\cygserver.exe' &) &&
+           "$WINDIR/System32/WindowsPowerShell/v1.0/powershell.exe" -command '& .\cygserver.exe' &) &&
           (cd $WINSUP && make check AM_COLOR_TESTS=always)
 
       - name: Upload testsuite logs
